### PR TITLE
Upgrade to a stable Lucene 7.5.0 release

### PR DESCRIPTION
**Pull request by jimczi**

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -178,7 +178,6 @@ if (project != rootProject) {
             ).asPath,
     )
     systemProperty 'test.local-test-repo-path', "${rootProject.buildDir}/local-test-repo"
-    systemProperty 'test.lucene-snapshot-revision', (versions.lucene =~ /\w+-snapshot-([a-z0-9]+)/)[0][1]
```
Comment (jimczi) [inline buildSrc/build.gradle]: @atorok what should we do with this line ? It fails if I keep it in this branch because we are now on a stable release.

**Review by jpountz — APPROVED** [review]
Let's get the issue you pinged @atorok about resolved, otherwise LGTM.

File: buildSrc/build.gradle (hunk 1 above)
Comment (atorok) [reply buildSrc/build.gradle]: @jimczi I left a time bomb. Sorry for that. I'll post  a patch here soon.

Comment (atorok) [reply buildSrc/build.gradle]:
```
diff --git a/buildSrc/build.gradle b/buildSrc/build.gradle
index b3cea1144a5..1310a906f64 100644
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -19,6 +19,9 @@
 import java.nio.file.Files
 import org.gradle.util.GradleVersion
 
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
 plugins {
   id 'java-gradle-plugin'
   id 'groovy'
@@ -178,6 +181,10 @@ if (project != rootProject) {
             ).asPath,
     )
     systemProperty 'test.local-test-repo-path', "${rootProject.buildDir}/local-test-repo"
+    Matcher isLuceneSnapshot = (/\w+-snapshot-([a-z0-9]+)/ =~ versions.lucene)
+    if (isLuceneSnapshot) {
+      systemProperty 'test.lucene-snapshot-revision', isLuceneSnapshot[0][1]
+    }
   }
   check.dependsOn(integTest)
 
diff --git a/buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java b/buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java
index aca99067011..2a2304182c8 100644
--- a/buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java
+++ b/buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java
@@ -117,7 +117,7 @@ public class BuildExamplePluginsIT extends GradleIntegrationTestCase {
         // Add a repositories section to be able to resolve dependencies
         String luceneSnapshotRepo = "";
         String luceneSnapshotRevision = System.getProperty("test.lucene-snapshot-revision");
-        if (luceneSnapshotRepo != null) {
+        if (luceneSnapshotRevision != null) {
             luceneSnapshotRepo =  "  maven {\n" +
                 "    url \"http://s3.amazonaws.com/download.elasticsearch.org/lucenesnapshots/" + luceneSnapshotRevision + "\"\n" +
                 "  }\n";
```

Comment (atorok) [reply buildSrc/build.gradle]: I pushed the fix to atorok@43bd5c7f347

Comment (jimczi) [reply buildSrc/build.gradle]: Thanks @atorok, I integrated the fix

**Review by alpar-t — APPROVED** [review]

**Comment by jimczi** [comment]
Thanks @atorok and @jpountz !

**Pull request merged**